app/models/wpcc/contributions_calendar.rb

```diff
@@ -0,0 +1,56 @@
+module Wpcc
+  class ContributionsCalendar
+    attr_reader :eligible_salary,
+                :salary_frequency,
+                :employee_percent,
+                :employer_percent
+
+    PERIODS_FILE = Wpcc::Engine.root.join('config', 'contribution_periods.yml')
+    PERIODS = YAML.load_file(PERIODS_FILE)
+
+    def initialize(eligible_salary:,
+                   employee_percent:,
+                   employer_percent:,
+                   salary_frequency:)
+
+      @eligible_salary = eligible_salary
+      @employee_percent = employee_percent
+      @employer_percent = employer_percent
+      @salary_frequency = salary_frequency
+    end
+
+    def schedule
```

Comment:
Actually the objective to we have the yaml file was:

1) The ContributionsCalendar class and Period Contribution Calculator **should not** know about any period that the tool implies.
2) The yaml will have the periods and some presenter class will handle your suggestions.

So, thank you for the comment, but I think this idea will tightly couple the code for presentation with the business logic.